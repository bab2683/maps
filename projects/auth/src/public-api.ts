/*
 * Public API Surface of auth
 */

export * from './lib/auth.model';
export * from './lib/auth.module';
export * from './lib/store';
