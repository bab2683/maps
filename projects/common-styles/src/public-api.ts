/*
 * Public API Surface of common-styles
 */
export * from './lib/common-styles.module';
