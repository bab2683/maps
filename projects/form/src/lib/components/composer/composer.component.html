<form [formGroup]="formGroup" *ngIf="formGroup" (ngSubmit)="submit()">
  <bab-form-row [data]="row" *ngFor="let row of formRows">
    <div class="field" *ngFor="let field of row.fields">
      <input class="field__input" [formControlName]="field.name" [type]="field.type" />
      <label class="field__label" [for]="field.name" *ngIf="field.label">{{ field.label }}</label>
    </div>
  </bab-form-row>
</form>
