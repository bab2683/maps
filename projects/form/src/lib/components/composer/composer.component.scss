@import "common";

input {
  &[type="email"],
  &[type="password"],
  &[type="text"] {
    background: none;
    border: 0;
    border-bottom: 2px solid $base-color-60;
    display: inline-block;
    padding: px-to-rem(10) 0 px-to-rem(2) px-to-rem(2);
    width: 100%;

    @extend .fz;

    @include transition(border-bottom-color);

    &:hover,
    &:focus {
      border-bottom-color: $base-color-90;
    }
  }
}

$field-class: ".field";

@mixin label-effect {
  transform: translateY(-40%);

  @extend .fz--small;
}

#{$field-class} {
  position: relative;

  &:hover {
    #{$field-class}__label {
      @include label-effect;
    }
  }

  &__label {
    left: 0;
    position: absolute;
    top: 0;
    transform: translateY(30%);
    width: 100%;

    @include transition((transform, font-size));
  }

  &__input {
    position: relative;

    &.ng-dirty:not(.ng-invalid) + #{$field-class}__label,
    &:focus + #{$field-class}__label {
      @include label-effect;
    }
  }
}
