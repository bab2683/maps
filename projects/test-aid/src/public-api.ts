/*
 * Public API Surface of test-aid
 */

export * from './lib/angular-fire.mock';
export * from './lib/rxjs.mock';
export * from './lib/utils.mock';
