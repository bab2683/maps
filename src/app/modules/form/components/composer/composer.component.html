<form [formGroup]="formGroup" *ngIf="formGroup" (ngSubmit)="submit()">
  <form-row [data]="row" *ngFor="let row of formRows">
    <form-field [data]="field" *ngFor="let field of row.fields">
      <label [for]="name" *ngIf="field.label">{{ label }}</label>
      <input
        [formControlName]="field.name"
        [attr.placeholder]="field.placeholder ? field.placeholder : null"
        [type]="field.type"
      />
    </form-field>
  </form-row>
</form>
