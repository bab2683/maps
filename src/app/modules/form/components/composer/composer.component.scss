@import "common";

input {
  &[type="email"],
  &[type="password"],
  &[type="text"] {
    background: none;
    border: 0;
    border-bottom: 2px solid $base-60;
    display: inline-block;
    padding: $base-spacing;
    width: 100%;

    @include transition(border-bottom-color);

    &:hover,
    :focus {
      border-bottom-color: $base-90;
    }
  }
}
