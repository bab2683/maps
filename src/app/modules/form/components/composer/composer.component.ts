import { Component, Input, OnInit } from '@angular/core';
import { FormBuilder, /*FormControl*/ FormGroup /*Validators*/ } from '@angular/forms';

import { FormField, FormResult, FormRow, ParsedData } from '../../interfaces';

@Component({
  selector: 'bab-form',
  templateUrl: './composer.component.html',
  styleUrls: ['./composer.component.scss']
})
export class ComposerComponent implements OnInit {
  public formFields: FormField[];
  public formRows: FormRow[];
  public formGroup: FormGroup;

  @Input()
  public rows: FormRow[];

  constructor(private fb: FormBuilder) {}

  public ngOnInit(): void {
    if (this.isDataValid()) {
      this.formRows = [...this.rows];
      this.initForm(this.parseData(this.formRows));
    }
  }

  public submit(): FormResult | null {
    if (this.formGroup.valid) {
      return this.formGroup.value;
    }
    return null;
  }

  private isDataValid(): boolean {
    return true;
  }

  private parseData(rows: FormRow[]): ParsedData {
    this.formFields = rows.reduce((fields, current) => {
      fields = [...fields, ...current.fields];
      return fields;
    }, []);

    return this.formFields.reduce<ParsedData>((parsedData, current) => {
      parsedData[current.name] = [
        current.default !== undefined ? current.default : null,
        current.validators ? current.validators : []
      ];
      return parsedData;
    }, {});
  }

  private initForm(parsedData: ParsedData): void {
    this.formGroup = this.fb.group(parsedData);
    console.log(this.formGroup);
  }
}
