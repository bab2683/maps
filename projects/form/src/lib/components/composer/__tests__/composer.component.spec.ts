import { NO_ERRORS_SCHEMA } from '@angular/compiler/src/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';
import { FormBuilder, FormsModule, ReactiveFormsModule } from '@angular/forms';
import { ComposerComponent } from '../composer.component';

describe('ComposerComponent', () => {
  let component: ComposerComponent;
  let fixture: ComponentFixture<ComposerComponent>;
  const formBuilder: FormBuilder = new FormBuilder();

  const testData = [
    {
      fields: [
        {
          label: 'Your email',
          name: 'email',
          type: 'email'
        },
        {
          label: 'Your password',
          name: 'password',
          type: 'password'
        }
      ]
    }
  ];

  let fbGroupSpy: any;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ComposerComponent],
      imports: [FormsModule, ReactiveFormsModule],
      providers: [
        {
          provide: FormBuilder,
          useValue: formBuilder
        }
      ],
      schemas: [NO_ERRORS_SCHEMA]
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ComposerComponent);
    component = fixture.componentInstance;
    fbGroupSpy = jest.spyOn(formBuilder, 'group');
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('onInit', () => {
    it('should create the form with formbuilder', () => {
      component.rows = testData;
      fixture.detectChanges();
      component.ngOnInit();
      expect(fbGroupSpy).toHaveBeenCalled();
    });
  });

  describe('submit', () => {
    beforeEach(() => {
      component.rows = testData;
      fixture.detectChanges();
      component.ngOnInit();
    });

    it('should return an object with values if form is valid', () => {
      // Null because I'm forcing the behavior
      const expectedResponse: { email: any; password: any } = { email: null, password: null };

      jest.spyOn(component.formGroup, 'valid', 'get').mockReturnValue(true);
      const result = component.submit();
      expect(result).toEqual(expectedResponse);
    });
    it('should return null if form is not valid', () => {
      const expectedResponse: any = null;

      jest.spyOn(component.formGroup, 'valid', 'get').mockReturnValue(false);
      const result = component.submit();
      expect(result).toEqual(expectedResponse);
    });
  });
});
