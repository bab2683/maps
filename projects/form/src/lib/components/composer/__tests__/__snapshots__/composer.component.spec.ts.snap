// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComposerComponent should match snapshot 1`] = `
<bab-form
  fb={[Function FormBuilder]}
>
  
</bab-form>
`;
